pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -109,10 +116,11 @@ func NewStaticPolicy(topology *topology.CPUTopology, numReservedCPUs int, reserv
 	klog.InfoS("Reserved CPUs not available for exclusive assignment", "reservedSize", reserved.Size(), "reserved", reserved)
 
 	return &staticPolicy{
-		topology:    topology,
-		reserved:    reserved,
-		affinity:    affinity,
-		cpusToReuse: make(map[string]cpuset.CPUSet),
+		topology:            topology,
+		reserved:            reserved,
+		affinity:            affinity,
+		cpusToReuse:         make(map[string]cpuset.CPUSet),
+		smtAwarenessEnabled: isSMTAwarePolicyOptionEnabled(cpuPolicyOptions),
 	}, nil
```

Comment:
Assuming we introduce an `Options` struct, as suggested above -- I think we should have a Constructor for it that takes in the options in `[]string` form and returns a populated `Options` struct that can be assigned here.

Likewise, this constructor should also return an error indicating whether the incoming list of strings are valid options or not.

Something like:
```
type Options struct {
    smtAwareness bool
    ...
}

func NewOptions([]string{}) (Options, error) {
   ...
}
```